repo/blob/azure/azure_storage.go

```diff
@@ -258,38 +256,43 @@ func (az *azStorage) putBlob(ctx context.Context, b blob.ID, data blob.Bytes, op
 // This protection is then removed and the main blob is deleted. Finally, the delete marker version is also deleted.
 // The original blob version protected by the policy is still protected from permanent deletion until the period has passed.
 func (az *azStorage) retryDeleteBlob(ctx context.Context, b blob.ID) error {
-	resp, err := az.putBlob(ctx, b, gather.FromSlice([]byte(deleteMarkerContent)), blob.PutOptions{
+	blobName := az.getObjectNameString(b)
+	resp, err := az.putBlob(ctx, b, gather.FromSlice([]byte(nil)), blob.PutOptions{
 		RetentionMode:   blob.RetentionMode(azblobblob.ImmutabilityPolicySettingUnlocked),
 		RetentionPeriod: time.Minute,
 	})
 	if err != nil {
-		return errors.Wrap(err, "failed to put delete marker blob version")
+		return errors.Wrap(err, "failed to put blob version needed to create delete marker")
 	}
 
 	_, err = az.service.ServiceClient().
 		NewContainerClient(az.container).
-		NewBlobClient(az.getObjectNameString(b)).
+		NewBlobClient(blobName).
 		DeleteImmutabilityPolicy(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "failed to remove delete marker blob immutability protection")
+		return errors.Wrap(err, "failed to create delete marker for immutable blob")
 	}
 
-	_, err = az.service.DeleteBlob(ctx, az.container, az.getObjectNameString(b), nil)
+	_, err = az.service.DeleteBlob(ctx, az.container, blobName, nil)
 	if err != nil {
 		return errors.Wrap(err, "failed to soft delete blob")
 	}
 
+	if resp.VersionID == nil || *resp.VersionID == "" {
+		// shouldn't happen
```

Comment:
log it? it shouldn't happen, that does not mean it won't happen.

Reply:
Added a log although I'm not sure if they're readable in some way (e.g. if the call stack higher up already has "maintenance" log